src/widgets/dialogs/EmotePopup.cpp

```diff
@@ -313,29 +354,119 @@ void EmotePopup::loadChannel(ChannelPtr _channel)
     }
 }
 
-void EmotePopup::loadEmojis()
+void EmotePopup::loadEmojis(ChannelView &view, EmojiMap &emojiMap)
 {
-    auto &emojis = getApp()->emotes->emojis.emojis;
-
     ChannelPtr emojiChannel(new Channel("", Channel::Type::None));
+    emojiChannel->addMessage(makeEmojiMessage(emojiMap));
+
+    view.setChannel(emojiChannel);
+}
+
+void EmotePopup::loadEmojis(Channel &channel, EmojiMap &emojiMap,
+                            const QString &title)
+{
+    channel.addMessage(makeTitleMessage(title));
+    channel.addMessage(makeEmojiMessage(emojiMap));
+}
+
+void EmotePopup::filterEmotes(const QString &text)
+{
+    if (text.length() > 0)
```

Comment:
Given that this only has two states, consider switching the logic around to look something like this instead:
```
if (text.empty()) {
    this->notebook_->show();
    this->searchView_->hide();

    return;
}

// search logic here
```

In addition to that, consider changing the argument name `text` to something more descriptive like `searchText` or `searchQuery`